server/src/main/java/org/opensearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -549,7 +574,9 @@ public void handleException(TransportException exp) {
                         retryTimeout,
                         ReplicationProxyFactory.create(
                             primaryShardReference.indexShard,
-                            getReplicationMode(primaryShardReference.indexShard)
+                            getReplicationMode(primaryShardReference.indexShard),
+                            replicasProxy,
+                            primaryTermValidationProxy()
```

Comment:
Maybe drop the ReplicationProxyFactory and directly inject either `ReplicationModeAwareProxy` or `FanoutReplicationProxy` with an inline check

Reply:
Ack. will do that.